### ISSUE TITLE
Put local images in custom welcome page

### ISSUE THREAD

**Issue by EliottPaillard** [ISAITB/gitb#85]
Hey @costas80,

We used already the customization feature of ITB pages but we didn't find how insert local images easily.
Is it possible to point to local files or do they need to be published online before?

**Comment by costas80** [comment]
Hi @EliottPaillard. I assume you mean the **community landing page** right (the first page you arrive to after logging in)?

If so, to add images you first need to add these as community resources. As the community admin go to your community's details and select **Resources**.

![image](https://github.com/user-attachments/assets/d3af7ab0-3183-47bc-b3e8-266c8dc8fb12)

Upload one or more resources that you want here (can be anything, but in your case it would be an image). Once you've defined your image as a resource, go to your landing page. You will notice there above the WYSIWYG editor a button **Copy resource reference**.

![image](https://github.com/user-attachments/assets/266f6fb8-6352-45c4-b79c-803c10884782)

With this you can search and find your image's resource. Selecting it will copy its path to the clipboard. You can then add this in the WYSIWYG editor as the source of an image. With the resource path copied, place the cursor where you want in the landing page and then click to add an image.

![image](https://github.com/user-attachments/assets/5fefe7b4-e4ac-438e-93e5-30d9f9b9e076)

In the **source** paste the copied resource reference. Doing so you will notice immediately that you get the image's dimensions.

![image](https://github.com/user-attachments/assets/7acd9709-ebee-442b-9752-66f3bb4a7b14)

Once you click on **save** the image will be added.

![image](https://github.com/user-attachments/assets/1ad4a0cc-45a2-4c57-b71a-64d3d9606dd7)

**Important note:** This image will only be visible to users within the community and are not accessible externally. In addition, if you use the Test Bed administrator account to view either these screens or the landing page, images added via community resources will appear broken. This is because to load such images the ITB needs to determine the user's community, which in the case of the Test Bed administrator is not directly possible. This is a known inconvenience that we want to address at some point. In your case however, if you just use community admins and users during your development you wont notice this.

**Tip:** You can use similarly community resources to add nice rich documentation for your test cases (e.g. UML diagrams), or to customise your certificate PDF reports. Also you can add anything as a resource - not just images. This allows you to e.g. also share like this archives with test data, documents etc. that you may not want to otherwise publish that you can add via links.

**Comment by EliottPaillard** [comment]
Great !
Yes, this is exactly what I was looking for.
Regarding the very first ITB page (before logging), is there also a way to add local images?

**Comment by costas80** [comment]
Currently no, as for these the ITB needs to know the user's community. To add a reference to a remote image log in as the Test Bed admin and adapt the welcome page message in the system configuration:

![image](https://github.com/user-attachments/assets/9d34355c-82b8-4bb5-aa29-23b4e52bfed9)

Regarding supporting local images, I'll add this to our backlog but I would treat it as low priority currently. Its quite niche and you should anyway be able to refer to a remote image if you are hosting the ITB (unlike a community's welcome page where you could have multiple communities in a single ITB instance). The same applies for images (logo, footer, favicon) you would be referring to when defining a [custom theme](https://www.itb.ec.europa.eu/docs/itb-ta/latest/systemAdministration/index.html#manage-themes).

**Comment by costas80** [comment]
> I'll add this to our backlog but I would treat it as low priority currently.

After checking this further I believe we could include the management and use of system-wide, public resources (that could be used on e.g. the welcome page or a default landing page) in our next release. I suggest we keep this issue open and resolve it once this feature is published in a nightly build.

**Comment by EliottPaillard** [comment]
> Regarding supporting local images, I'll add this to our backlog but I would treat it as low priority currently. Its quite niche and you should anyway be able to refer to a remote image if you are hosting the ITB (unlike a community's welcome page where you could have multiple communities in a single ITB instance). The same applies for images (logo, footer, favicon) you would be referring to when defining a [custom theme](https://www.itb.ec.europa.eu/docs/itb-ta/latest/systemAdministration/index.html#manage-themes).

Yes of course, we don't really need it for the time being anyway.
But it would be good to have this possibility later.

**Comment by costas80** [comment]
Hi there @EliottPaillard!

The feature we were discussing is now available on our **nightly channel**. The Test Bed administrator can now upload system-wide resources (same as [community resources](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#manage-resources) but system-wide) that are publicly accessible to all users (also users without an account on the Test Bed instance in question). Such resources can be used everywhere you can define rich text:

- In system and community-specific [landing pages](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#manage-landing-pages), [legal notices](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#manage-legal-notices) and [error templates](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#manage-error-templates).
- In [test suite](https://www.itb.ec.europa.eu/docs/itb-ta/latest/domainDashboard/index.html#manage-test-suite-details), [test case](https://www.itb.ec.europa.eu/docs/itb-ta/latest/domainDashboard/index.html#manage-test-case-details) and [test step](https://www.itb.ec.europa.eu/docs/tdl/latest/constructs/index.html#rich-documentation-per-step) documentation.
- In custom messages for PDF conformance certificates (at [statement](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#conformance-statement-certificate) and [overview](https://www.itb.ec.europa.eu/docs/itb-ta/latest/communityDashboard/index.html#conformance-overview-certificate) levels). 
- In the Test Bed's [welcome page](https://www.itb.ec.europa.eu/docs/itb-ta/latest/login/index.html#welcome-page).

As an example, and to revisit the case we were discussing in previous comments, to add an image (or e.g. link to a resource) on the welcome page you do as follows. First connect as the Test Bed administrator and navigate to **System administration**. You will see in the bottom panel a new tab named **Resources**. This is where you upload system-wide resources (exactly as you would for community-specific ones):

![Image](https://github.com/user-attachments/assets/ce5e51b3-593b-4dde-880e-dc34fd2b1d80)

Having uploaded an image here, you can then expand the **Custom welcome page message** setting under **System configuration**. Doing so you will notice a **Copy system-wide resource reference** selection button to list the resources and copy the reference for the one you want (e.g. an image). You can then add an image in the editor based on the reference you copied.

![Image](https://github.com/user-attachments/assets/7dad1e08-1c82-4a9f-a291-0c310df00d24)

Having saved the adapted message, you will now see the update when accessing the welcome page:

![Image](https://github.com/user-attachments/assets/dc780066-1e2e-4fc6-b615-dd4c6db4f6a9)

I assume this covers your needs no? As always, happy to hear any feedback you may have.
